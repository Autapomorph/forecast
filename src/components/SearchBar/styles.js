import styled, { css } from 'styled-components';

export const StyledSearchForm = styled.form`
  display: flex;
  width: 100%;
  padding: 0 10px;
`;

export const StyledSearchInput = styled.input.attrs({
  type: 'text',
})`
  flex-grow: 1;
  background: var(--white);
  padding: 5px;
  border: 1px solid var(--gray);
  border-radius: 5px 0 0 5px;
  font-size: 1.5rem;
  outline: none;
`;

export const StyledSearchButton = styled.button.attrs({
  type: 'submit',
})`
  flex-shrink: 0;
  padding: 5px 10px;
  border: 1px solid var(--gray);
  border-left: 0;
  border-radius: 0 5px 5px 0;
  background: var(--white);
  color: var(--base-text-color);
  font-size: 1.5rem;
  cursor: pointer;

  &:hover {
    background: var(--light-gray);
    color: var(--base-text-color);
  }

  ${({ disabled }) =>
    disabled &&
    css`
      background: var(--light-gray);
      color: var(--disabled-text-color);
      cursor: default;

      &:hover {
        color: var(--disabled-text-color);
      }
    `};
`;
