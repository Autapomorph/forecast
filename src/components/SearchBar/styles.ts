import styled, { css } from 'styled-components/macro';

import { ThemeProp } from 'models';

export const SearchForm = styled.form`
  display: flex;
  width: 100%;
  padding: 0 10px;
  margin-bottom: 15px;
`;

export const SearchInput = styled.input.attrs({
  type: 'text',
})`
  flex-grow: 1;
  background: ${({ theme }: ThemeProp) => theme.inputBg};
  color: ${({ theme }: ThemeProp) => theme.inputColor};
  padding: 10px;
  border: 1px solid ${({ theme }: ThemeProp) => theme.borderColor};
  border-right: 0;
  border-radius: 5px 0 0 5px;
  font-size: 1.3rem;
  line-height: 1.3;
  outline: none;

  ::placeholder {
    color: ${({ theme }: ThemeProp) => theme.inputPlaceholderColor};
  }

  ::selection {
    background: ${({ theme }: ThemeProp) => theme.inputSelectionBg};
    color: ${({ theme }: ThemeProp) => theme.inputSelectionColor};
  }
`;

export const InputButtonsBlock = styled.div`
  display: flex;
  flex-direction: row;
  flex-shrink: 0;
`;

export const InputButton = styled.button.attrs({
  type: 'button',
})`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-shrink: 0;
  padding: 5px 10px;
  border: none;
  border-right: 1px solid ${({ theme }: ThemeProp) => theme.menuItemBorderColor};
  background: ${({ theme }: ThemeProp) => theme.menuItemBg};
  color: ${({ theme }: ThemeProp) => theme.menuItemColor};
  font-size: 1.5rem;
  transition: 0.3s;
  cursor: pointer;

  :last-child {
    border: none;
    border-radius: 0 5px 5px 0;
  }

  :hover {
    background: ${({ theme }: ThemeProp) => theme.menuItemHoverBg};
  }

  ${({ disabled }) =>
    disabled &&
    css`
      background: ${({ theme }: ThemeProp) => theme.menuItemDisabledBg};
      color: ${({ theme }: ThemeProp) => theme.menuItemDisabledColor};
      cursor: default;

      :hover {
        background: ${({ theme }: ThemeProp) => theme.menuItemDisabledBg};
        color: ${({ theme }: ThemeProp) => theme.menuItemDisabledColor};
      }
    `}
`;

export const SettingsButton = styled(InputButton)``;

export const SearchButton = styled(InputButton).attrs({
  type: 'submit',
})``;

export const ClearButton = styled(InputButton)`
  border: none;
  border-top: 1px solid ${({ theme }: ThemeProp) => theme.borderColor};
  border-bottom: 1px solid ${({ theme }: ThemeProp) => theme.borderColor};
  background: ${({ theme }: ThemeProp) => theme.menuClearButtonBg};
  color: ${({ theme }: ThemeProp) => theme.menuClearButtonColor};
  outline: none;
  transition: none;

  :hover {
    background: ${({ theme }: ThemeProp) => theme.menuClearButtonBg};
    color: ${({ theme }: ThemeProp) => theme.menuClearButtonHoverColor};
  }

  ${({ disabled }) =>
    disabled &&
    css`
      cursor: default;
    `}

  svg {
    opacity: 1;
    transition: 0.3s;

    ${({ disabled }) =>
      disabled &&
      css`
        opacity: 0;
      `}
  }
`;
