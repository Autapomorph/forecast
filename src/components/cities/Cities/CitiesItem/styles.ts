import styled from 'styled-components/macro';

export const CitiesItemWrapper = styled.li`
  padding: 10px;
  margin-bottom: 15px;
  border: 1px solid var(--light-gray);
  border-radius: 5px;
  background: var(--white);
  color: var(--contrast-text-color);
  font-weight: bold;

  :last-child {
    margin-bottom: 0;
  }
`;

export const CitiesItemTitle = styled.span`
  cursor: pointer;
`;

export const TitleButtonsWrapper = styled.div`
  flex-shrink: 0;

  > :not(:last-child) {
    margin-right: 10px;
  }
`;

export const CitiesItemHeader = styled.h4`
  display: flex;
  align-items: center;
  margin: 5px 0;
  font-weight: bold;

  ${CitiesItemTitle} {
    flex-grow: 1;
  }
`;

export const CitiesItemContent = styled.p`
  margin: 10px 0 0 0;
`;
