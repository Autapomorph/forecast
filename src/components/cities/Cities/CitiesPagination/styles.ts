import styled, { css } from 'styled-components/macro';

type Props = {
  isSinglePage: boolean;
};

export const CitiesPagination = styled.div`
  .pagination {
    display: flex;
    justify-content: flex-end;
    list-style: none;

    ${({ isSinglePage }: Props) =>
      isSinglePage &&
      css`
        display: none;
      `}
  }

  .page {
    background: var(--white);
    color: var(--contrast-text-color);
    transition: 0.3s;
    cursor: pointer;

    :hover {
      background: var(--gray);
    }

    &--prev {
      border-radius: 5px 0 0 5px;
    }

    &--next {
      border-radius: 0 5px 5px 0;
    }

    &--disabled {
      background: var(--light-gray);
      color: var(--disabled-text-color);
      cursor: default;

      :hover {
        background: var(--light-gray);
        color: var(--disabled-text-color);
      }
    }

    &--hidden {
      display: none;
    }

    a {
      display: block;
      padding: 10px;
      outline: none;
    }
  }
`;
