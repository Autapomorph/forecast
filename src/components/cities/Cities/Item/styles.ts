import styled from 'styled-components/macro';

import { ThemeProp } from 'models';

export const CitiesItemWrapper = styled.li`
  display: flex;
  min-height: 50px;
  margin-bottom: 15px;
  border: 1px solid ${({ theme }: ThemeProp) => theme.borderColor};
  border-radius: 5px;
  background: ${({ theme }: ThemeProp) => theme.elementBg};
  color: ${({ theme }: ThemeProp) => theme.altTextColor};
  font-weight: bold;

  :last-child {
    margin-bottom: 0;
  }
`;

export const CitiesItemTitle = styled.span`
  align-self: center;
  flex-grow: 1;
  padding: 15px 0;
  padding-left: 10px;
  cursor: pointer;
`;

export const TitleButtonsWrapper = styled.div`
  display: flex;
  flex-shrink: 0;
  padding-left: 10px;
  padding-right: 5px;

  > :not(:last-child) {
    margin-right: 10px;
  }
`;
