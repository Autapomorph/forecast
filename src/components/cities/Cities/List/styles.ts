import styled from 'styled-components/macro';

export const CitiesList = styled.ul`
  padding: 0;
  margin: 12px 0 0;
  list-style: none;
`;
