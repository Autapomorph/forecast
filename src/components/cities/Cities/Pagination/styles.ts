import styled, { css } from 'styled-components/macro';

import { ThemeProp } from 'models';

type Props = {
  isSinglePage: boolean;
};

export const CitiesPagination = styled.div`
  .pagination {
    display: flex;
    justify-content: flex-end;
    list-style: none;

    ${({ isSinglePage }: Props) =>
      isSinglePage &&
      css`
        display: none;
      `}
  }

  .page {
    background: ${({ theme }: ThemeProp) => theme.paginationBg};
    color: ${({ theme }: ThemeProp) => theme.paginationColor};
    transition: 0.3s;
    cursor: pointer;

    :hover {
      background: ${({ theme }: ThemeProp) => theme.paginationHoverBg};
    }

    &--prev {
      border-radius: 5px 0 0 5px;
    }

    &--next {
      border-radius: 0 5px 5px 0;
    }

    &--disabled {
      background: ${({ theme }: ThemeProp) => theme.paginationDisabledBg};
      color: ${({ theme }: ThemeProp) => theme.paginationDisabledColor};
      cursor: default;

      :hover {
        background: ${({ theme }: ThemeProp) => theme.paginationDisabledBg};
        color: ${({ theme }: ThemeProp) => theme.paginationDisabledColor};
      }
    }

    &--hidden {
      display: none;
    }

    a {
      display: block;
      padding: 10px;
      outline: none;
    }
  }
`;
