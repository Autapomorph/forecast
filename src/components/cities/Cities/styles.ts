import styled from 'styled-components/macro';

type Props = {
  isLoading: boolean;
};

export const CitiesSection = styled.section`
  grid-area: cities;
  padding: 0 10px;
  margin-bottom: ${({ isLoading }: Props) => (isLoading ? '0' : '20px')};
`;

export const CitiesHeader = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`;
