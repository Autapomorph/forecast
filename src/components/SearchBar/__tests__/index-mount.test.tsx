import React from 'react';
import { mount } from 'enzyme';

import { SearchBar } from '..';

import * as S from '../styles';

jest.mock('components/SearchBar/GeolocationButton', () => () => null);

describe('SearchBar tests', () => {
  const fetchCititesByName = jest.fn();
  const props = {
    _fetchCititesByName: fetchCititesByName,
  };

  it('renders properly', () => {
    const wrapper = mount(<SearchBar {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should focus input after ClearButton click', () => {
    const wrapper = mount(<SearchBar {...props} />);
    wrapper.find(S.SearchInput).invoke('onChange')({
      target: { value: 'cityName' },
    });
    wrapper.find(S.ClearButton).invoke('onClick')();

    expect(wrapper.find('input').is(':focus')).toBeTruthy();
  });

  it('should blur input after submit', () => {
    const wrapper = mount(<SearchBar {...props} />);
    wrapper.find(S.SearchInput).invoke('onChange')({
      target: { value: 'cityName' },
    });
    wrapper.find(S.ClearButton).invoke('onClick')();
    wrapper.find(S.SearchForm).invoke('onSubmit')({ preventDefault: jest.fn() });

    expect(wrapper.find('input').is(':focus')).toBeFalsy();
  });
});
