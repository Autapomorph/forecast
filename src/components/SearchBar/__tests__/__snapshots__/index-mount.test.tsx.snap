// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar tests renders properly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  padding: 0 10px;
  margin-bottom: 15px;
}

.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  padding: 10px;
  border: 1px solid;
  border-right: 0;
  border-radius: 5px 0 0 5px;
  font-size: 1.3rem;
  line-height: 1.3;
  outline: none;
}

.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  padding: 10px;
  border: 1px solid;
  border-right: 0;
  border-radius: 5px 0 0 5px;
  font-size: 1.3rem;
  line-height: 1.3;
  outline: none;
}

.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  padding: 10px;
  border: 1px solid;
  border-right: 0;
  border-radius: 5px 0 0 5px;
  font-size: 1.3rem;
  line-height: 1.3;
  outline: none;
}

.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  padding: 10px;
  border: 1px solid;
  border-right: 0;
  border-radius: 5px 0 0 5px;
  font-size: 1.3rem;
  line-height: 1.3;
  outline: none;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding: 5px 10px;
  border: none;
  border-right: 1px solid;
  font-size: 1.5rem;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  cursor: pointer;
}

.c4:last-child {
  border: none;
  border-radius: 0 5px 5px 0;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding: 5px 10px;
  border: none;
  border-right: 1px solid;
  font-size: 1.5rem;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  cursor: pointer;
  cursor: default;
}

.c5:last-child {
  border: none;
  border-radius: 0 5px 5px 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding: 5px 10px;
  border: none;
  border-right: 1px solid;
  font-size: 1.5rem;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  cursor: pointer;
  cursor: default;
  border: none;
  border-top: 1px solid;
  border-bottom: 1px solid;
  outline: none;
  -webkit-transition: none;
  transition: none;
  cursor: default;
}

.c2:last-child {
  border: none;
  border-radius: 0 5px 5px 0;
}

.c2 svg {
  opacity: 1;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  opacity: 0;
}

<SearchBar
  _fetchCititesByName={[MockFunction]}
>
  <styles__SearchForm
    onSubmit={[Function]}
  >
    <form
      className="c0"
      onSubmit={[Function]}
    >
      <styles__SearchInput
        onChange={[Function]}
        placeholder="City"
        value=""
      >
        <input
          className="c1"
          onChange={[Function]}
          placeholder="City"
          type="text"
          value=""
        />
      </styles__SearchInput>
      <styles__ClearButton
        disabled={true}
        onClick={[Function]}
      >
        <button
          className="styles__InputButton-sc-9djs8t-3 c2"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  352,
                  512,
                  Array [],
                  "f00d",
                  "M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z",
                ],
                "iconName": "times",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-times fa-w-11 "
              data-icon="times"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 352 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </FontAwesomeIcon>
        </button>
      </styles__ClearButton>
      <styles__InputButtonsBlock>
        <div
          className="c3"
        >
          <styles__SettingsButton
            disabled={false}
            onClick={[Function]}
          >
            <button
              className="styles__InputButton-sc-9djs8t-3 c4"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      512,
                      512,
                      Array [],
                      "f1de",
                      "M496 384H160v-16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v16H16c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h80v16c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-16h336c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zm0-160h-80v-16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v16H16c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h336v16c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-16h80c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zm0-160H288V48c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v16H16C7.2 64 0 71.2 0 80v32c0 8.8 7.2 16 16 16h208v16c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-16h208c8.8 0 16-7.2 16-16V80c0-8.8-7.2-16-16-16z",
                    ],
                    "iconName": "sliders-h",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                swapOpacity={false}
                symbol={false}
                title=""
                transform={null}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-sliders-h fa-w-16 "
                  data-icon="sliders-h"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M496 384H160v-16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v16H16c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h80v16c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-16h336c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zm0-160h-80v-16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v16H16c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h336v16c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-16h80c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zm0-160H288V48c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v16H16C7.2 64 0 71.2 0 80v32c0 8.8 7.2 16 16 16h208v16c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-16h208c8.8 0 16-7.2 16-16V80c0-8.8-7.2-16-16-16z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </FontAwesomeIcon>
            </button>
          </styles__SettingsButton>
          <styles__SearchButton
            disabled={true}
          >
            <button
              className="styles__InputButton-sc-9djs8t-3 c5"
              disabled={true}
              type="submit"
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      512,
                      512,
                      Array [],
                      "f002",
                      "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                    ],
                    "iconName": "search",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                swapOpacity={false}
                symbol={false}
                title=""
                transform={null}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-search fa-w-16 "
                  data-icon="search"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </FontAwesomeIcon>
            </button>
          </styles__SearchButton>
          <Component
            onClick={[Function]}
          />
        </div>
      </styles__InputButtonsBlock>
    </form>
  </styles__SearchForm>
</SearchBar>
`;
